### PR TITLE
Fix: Button and add ButtonGroup

### PR DESCRIPTION
**Pull request by rohits-geekyants**

### CODE REVIEW

File: src/components/primitives/Button/index.tsx [hunk 1]
```diff
@@ -309,92 +327,104 @@ const Button = ({
...
+const applyProps = (children: any, props: any) => {
+  return children.map((child: JSX.Element) => {
+    return React.cloneElement(child, props, child.props.children);
+  });
+};
+
+export const ButtonGroup = ({ children, ...props }: ButtonGroupProps) => {
+  return (
+    <StyledButtonGroup>
+      {applyProps(getAttachedChildren(children), props)}
```
Comment (theankurkedia) [inline src/components/primitives/Button/index.tsx:425]: I am not sure that `applyProps` conveys the accurate functionality we are trying to achieve here. Can change it to something more apt.

File: src/theme/theme.ts [hunk 2]
```diff
@@ -14,7 +14,7 @@ export default {
     muted: ['#e2e3e5', '#65696d', '#acacac'],
     dark: ['#d5d8d9', '#1b1f22', '#004466'],
     light: ['#fefefe', '#818182', '#f8f9fb'],
-    default: ['#edf2f6', '#1a212c', '#718096'],
+    default: ['#c3dafe', '#434190', '#5a67d8'],
```
Comment (theankurkedia) [inline src/theme/theme.ts:17]: Why did we do this? This would affect all components, not just Button. Please justify this.

**Review by theankurkedia — CHANGES_REQUESTED** [review]
Please check the comments.

File: src/theme/theme.ts (hunk 2 above)
Comment (rohits-geekyants) [reply src/theme/theme.ts:17]: Previously we were having whitesmoke background and black color for text as default, I changed it to the more common and better looking primary blue.

File: src/components/primitives/Button/index.tsx (hunk 1 above)
Comment (rohits-geekyants) [reply src/components/primitives/Button/index.tsx:425]: supplyPropsToChildren would be ok?

Comment (theankurkedia) [reply src/components/primitives/Button/index.tsx:425]: Yes, that would be better.

File: src/theme/theme.ts (hunk 2 above)
Comment (theankurkedia) [reply src/theme/theme.ts:17]: Ideally, we should have another PR for this. But it's fine for now.

File: src/components/primitives/Button/index.tsx (hunk 1 above)
Comment (rohits-geekyants) [reply src/components/primitives/Button/index.tsx]: Updated

**Pull request merged**